Sources/ClientRuntime/Interceptor/DefaultInterceptorContext.swift

```diff
@@ -85,8 +93,8 @@ extension DefaultInterceptorContext: AfterAttempt {
 }
 
 extension DefaultInterceptorContext: MutableOutputAfterAttempt {
-    public func updateResult(updated: Result<OutputType, Error>) {
-        self.result = updated
+    public func updateOutput(updated: OutputType) {
+        self.result = .success(updated)
```

Comment:
before updated could be an `Error`, will this always be a success?

Reply:
The orchestrator uses [setResult](https://github.com/smithy-lang/smithy-swift/pull/704/files/157b12fcffaacb0145c94d499554e626f9f291f9#diff-8184a6ba179295649afb02c7f9473e695195634dfcc8b1ab425ca8d4c916b3c1R41) to set errors. So internally DefaultInterceptorContext keeps track of failures - if you look at `getOutput`, you'll see it now throws.

This change was based on discussion with @jbelkins about mixing and matching `Result` with `throws` in Swift.